logp/core.go

```diff
@@ -256,6 +290,36 @@ func ConfigureWithTypedOutput(defaultLoggerCfg, typedLoggerCfg Config, key, valu
 	return nil
 }
 
+func ConfigureWithTypedOutputLocal(defaultLoggerCfg, typedLoggerCfg Config, key, value string, outputs ...zapcore.Core) (*Logger, error) {
+	sink, level, _, selectors, err := createSink(defaultLoggerCfg, outputs...)
+	if err != nil {
+		return nil, err
+	}
+
+	var typedCore zapcore.Core
+	if defaultLoggerCfg.toObserver {
+		typedCore = sink
+	} else {
+		typedCore, err = createLogOutput(typedLoggerCfg, level)
+	}
+	if err != nil {
+		return nil, fmt.Errorf("could not create typed logger output: %w", err)
+	}
+
+	sink = &typedLoggerCore{
+		defaultCore: sink,
+		typedCore:   typedCore,
+		key:         key,
+		value:       value,
+	}
+
+	sink = selectiveWrapper(sink, selectors)
```

Comment:
Same here, it seems a duplication of most of the code from `ConfigureWithTypedOutput`. Ideally it'd be extracted to a unreported function and both `ConfigureWithTypedOutput` and `ConfigureWithTypedOutputLocal` would call it.